lib/index.js

```diff
@@ -96,16 +96,42 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
 
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
+  /*  subscription filter control */
+  this.channels     = {};
+  this.register     = register;
+
+  /*  subscription handling at initialization  */
+  if (opts.hasOwnProperty('chan')) {
+    if (Array.isArray(opts.chan)) {
+      opts.chan.forEach(register.bind(this));
+    } else {
+      throw new TypeError('chan requires an Array');
     }
+  } else if (type === 'sub') {
+    this.register(''); //default topic is an empty string
+  }
 
-    if(this.af_domain == nn.AF_SP) {
-        if (this.receiver) this._startPollReceive();
+  function register(chan) {
+    if (this.channels.hasOwnProperty('')) {
+      this.rmchan(''); return this.register(chan);
+    } else if (nn.Chan(this.binding, nn.NN_SUB_SUBSCRIBE, chan) !== -1) {
+      return this.channels[chan] = true;
     }
+    return this.emit('error', new Error( nn.Err() + ' : ' + chan));
+  }
```

Comment:
Can we make register a method of the prototype?
```js
Socket.prototype._register(chan) {};
```
Let's make each statement it's own line.
Let's remove return values since they're not used by any callers.
```js
if (this.chan...) {
} else if (nn.Chan) {
} else {}
// no explicit return statements
```

After this is fixed, the PR looks good to merge.  Now that Travis is hooked up, I trust it enough that I don't have to pull down the patch and test it locally! :rocket: :guitar: